integration_test.go

```diff
@@ -359,40 +361,386 @@ func setupYARPCServer(t *testing.T, inbound ytransport.Inbound, opts ...ythrift.
+					Procedure:         "Bar/ServerStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `test: 2`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+			},
+			wantRes: `{
+  "test": 1
+}
+
+{
+  "test": 2
+}`,
+			expectedInput: []simple.Foo{{Test: 2}},
+			returnOutput:  []simple.Foo{{Test: 1}, {Test: 2}},
+		},
+		{
+			desc: "bidirectional streaming",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/BidiStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{"test":250}{"test":1}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+			},
+			wantRes: `{
+  "test": 350
+}
+
+{
+  "test": 101
+}`,
+			expectedInput: []simple.Foo{{Test: 250}, {Test: 1}},
+			returnOutput:  []simple.Foo{{Test: 350}, {Test: 101}},
+		},
+		{
+			desc: "bidirectional streaming with YAML input",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/BidiStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON: `test: 250
+---
+test: 1
+---`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+			},
+			wantRes: `{
+  "test": 350
+}
+
+{
+  "test": 101
+}`,
+			expectedInput: []simple.Foo{{Test: 250}, {Test: 1}},
+			returnOutput:  []simple.Foo{{Test: 350}, {Test: 101}},
+		},
+		{
+			desc: "client streaming with invalid input",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/ClientStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+			},
+			wantRes: ``,
+			wantErr: "Failed while reading stream input: unexpected EOF\n",
+		},
+		{
+			desc: "bidirectional streaming with invalid second input",
+			opts: Options{
+				ROpts: RequestOptions{
+					FileDescriptorSet: []string{"testdata/protobuf/simple/simple.proto.bin"},
+					Procedure:         "Bar/BidiStream",
+					Timeout:           timeMillisFlag(time.Second),
+					RequestJSON:       `{"test":250}{"test_err": 1}`,
+				},
+				TOpts: TransportOptions{
+					ServiceName: "foo",
+				},
+			},
+			wantErr:       `could not parse given request body as message of type "Foo"`,
+			expectedInput: []simple.Foo{{Test: 250}, {Test: 1}},
+			returnOutput:  []simple.Foo{{Test: 350}, {Test: 101}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			addr, server, simpleSvc := setupGRPCServer(t)
+			defer server.Stop()
+
+			simpleSvc.returnError = tt.returnError
+			simpleSvc.returnOutput = tt.returnOutput
+			simpleSvc.expectedInput = tt.expectedInput
```

Comment:
optional/can refactor in another PR:
this assignment looks a bit weird, like it is has side effects on return values  in `setupGRPCServer`. Do you think the following might be more clear?

```go
func setupGRPCServer(t *testing.T, ss *simpleService) {
  if ss == nil {
    ss = &simpleService{}
  }
  ...
}
```
and here we can just do

```go
addr, server := setupGRPCServer(t, &simpleService{returnError: tt.returnError, // other 2 assignments})
```